lib/galaxy/tools/parameters/validation.py

```diff
@@ -58,7 +58,7 @@ def __init__( self, message, expression ):
         self.expression = expression
 
     def validate( self, value, trans=None ):
-        if re.match( self.expression, value ) is None:
+        if re.match( self.expression, value or '' ) is None:
```

Comment:
Wouldn't this break the validation of things that are not optional and that cannot be empty?